hw03_frequency_analysis/top.go

```diff
@@ -1,6 +1,90 @@
 package hw03frequencyanalysis
 
-func Top10(_ string) []string {
-	// Place your code here.
-	return nil
+import (
+	"errors"
+	"regexp"
+	"strings"
+	"unicode"
+)
+
+type Word struct {
+	Word  string
+	Count int
+}
+
+var re = regexp.MustCompile(`^[[:punct:]]+|[[:punct:]]+$`)
+
+func PrepareWord(w string) (string, error) {
+	if len(w) < 2 {
+		for _, s := range w {
+			if !unicode.IsLetter(s) {
+				return "", errors.New("is not word")
+			}
+		}
+	}
+	return strings.ToLower(re.ReplaceAllString(w, "")), nil
+}
+
+func Top10(s string) []string {
+	if s == "" {
+		return []string{}
+	}
+
+	uniqueWords := make(map[string]int)
+	words := strings.Fields(s)
+
+	for _, word := range words {
+		// проверить что слово является именно словом и отрезать от него знаки препинания, если они есть
+		prepWord, err := PrepareWord(word)
+		if err != nil {
+			continue
+		}
+
+		_, ok := uniqueWords[prepWord]
+
+		if !ok {
+			uniqueWords[prepWord] = 1
+		} else {
+			uniqueWords[prepWord]++
+		}
```

Comment:
```suggestion
		uniqueWords[prepWord]++
```
Так тоже будет работать